coderd/templateversions_test.go

```diff
@@ -128,6 +129,57 @@ func TestPostTemplateVersionsByOrganization(t *testing.T) {
 		require.Len(t, auditor.AuditLogs, 1)
 		assert.Equal(t, database.AuditActionCreate, auditor.AuditLogs[0].Action)
 	})
+	t.Run("Example", func(t *testing.T) {
+		t.Parallel()
+		client := coderdtest.New(t, nil)
+		user := coderdtest.CreateFirstUser(t, client)
+
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		ls, err := examples.List()
+		require.NoError(t, err)
+
+		// try a bad example ID
+		_, err = client.CreateTemplateVersion(ctx, user.OrganizationID, codersdk.CreateTemplateVersionRequest{
+			Name:          "my-example",
+			StorageMethod: codersdk.ProvisionerStorageMethodFile,
+			ExampleID:     "not a real ID",
+			Provisioner:   codersdk.ProvisionerTypeEcho,
+		})
+		require.Error(t, err)
+		require.ErrorContains(t, err, "not found")
+
+		// try file and example IDs
+		_, err = client.CreateTemplateVersion(ctx, user.OrganizationID, codersdk.CreateTemplateVersionRequest{
+			Name:          "my-example",
+			StorageMethod: codersdk.ProvisionerStorageMethodFile,
+			ExampleID:     ls[0].ID,
+			FileID:        uuid.New(),
+			Provisioner:   codersdk.ProvisionerTypeEcho,
+		})
+		require.Error(t, err)
+		require.ErrorContains(t, err, "example_id")
+		require.ErrorContains(t, err, "file_id")
+
+		// try a good example ID
+		tv, err := client.CreateTemplateVersion(ctx, user.OrganizationID, codersdk.CreateTemplateVersionRequest{
+			Name:          "my-example",
+			StorageMethod: codersdk.ProvisionerStorageMethodFile,
+			ExampleID:     ls[0].ID,
+			Provisioner:   codersdk.ProvisionerTypeEcho,
+		})
+		require.NoError(t, err)
+		require.Equal(t, "my-example", tv.Name)
+
+		// ensure the template tar was uploaded correctly
+		fl, ct, err := client.Download(ctx, tv.Job.FileID)
+		require.NoError(t, err)
+		require.Equal(t, "application/x-tar", ct)
+		tar, err := examples.Archive(ls[0].ID)
+		require.NoError(t, err)
+		require.EqualValues(t, tar, fl)
```

Comment:
Since `examples.Archive` generates an archive on the fly is there any chance that they will be generated and be slightly different?

Reply:
its the same underlying data source so I don't think so